middleware/src/main/java/br/ufrn/dimap/middleware/lifecycle/PoolingI.java

```diff
@@ -0,0 +1,26 @@
+package br.ufrn.dimap.middleware.lifecycle;
+
+/**
+ * This is an interface created for pooling pattern to be used by the
+ * Per-request instances.
+ *
+ * @author Gabriel Victor
+ * @version 1.0
+ */
+public interface PoolingI<T> {
```

Comment:
Consider to use the same pattern for names as the other classes and interfaces in this project, which is "interface Name" - instead of "interface NameI" - and  "class MoreSpecificName" (or "class NameImpl").